app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :name, presence: true,
+    length: {maximum: Settings.user.name.length}
+  validates :email, presence: true,
+    length: {maximum: Settings.user.email.length},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: Settings.user.case_sensitive}
+  validates :password, presence: true,
+    length: {minimum: Settings.user.password.length}
+
+  has_secure_password
```

Comment:
cái này nghĩa là gì nhỉ, tại sao lại để nó ở đây

Reply:
Phương thức has_secure_password để:
-load Gem bcrypt
-Thêm thuộc tính read-only cho password
-Mật khẩu được xác nhận bởi 1 trường khác(password_confirmation). Tức là phải nhập pass 2 lần
-Xác thực, nếu pass đúng thì trả về true, không thì false.
Đặt ở đây để đảm bảo an toàn cho mật khẩu.

Reply:
trong db đâu có field password đâu nhỉ :-? cái password này ở đâu ra thế =))